pkg/kvstore/etcd.go

```diff
@@ -1162,11 +1214,16 @@ func (e *etcdClient) determineEndpointStatus(ctx context.Context, endpointAddres
 
 	e.getLogger().Debugf("Checking status to etcd endpoint %s", endpointAddress)
 
-	e.limiter.Wait(ctxTimeout)
+	rl, err := e.limiter.Wait(ctxTimeout)
```

Comment:
just a small nit: is there a reason you are mixing lr/rl ? e.g. `rl, err := e.limiter.Wait(ctxTimeout)` and `lr, err := e.limiter.Wait(ctx)`